roles/openshift_persistent_volumes/action_plugins/generate_pv_pvcs_list.py

```diff
@@ -0,0 +1,159 @@
+"""
```

Comment:
I'm not a fan of the filter_plugin that was generating these and so I understand the desire to move this somewhere else.  I'm not sure I understand why this code is needed when we already have a module that does persistent volume claims, `lib_openshift/oc_pvc.py`.   This makes it difficult to track down what and how a pvc is created.  The module method looks like this:

```
- name: create prometheus pvc
  oc_pvc:
    namespace: "{{ openshift_prometheus_namespace }}"
    name: "{{ openshift_prometheus_pvc_name }}"
    access_modes: "{{ openshift_prometheus_pvc_access_modes }}"
    volume_capacity: "{{ openshift_prometheus_pvc_size }}"
    selector: "{{ openshift_prometheus_pvc_pv_selector }}"
  when: openshift_prometheus_storage_type == 'pvc'
```

Each role that requires a pvc should call the module rather than having a single place that generates all of the persistent volume claims at once.  That way every role and technology that is getting deployed can stand on its own.  Not to mention the module gives us an ansible interface for creating oc objects it also gives us idempotency.

I won't block on this because we are moving in the right direction with this PR by reducing facts but we should also remove this code (less to maintain, debug, etc) and move in the direction of a module.

Reply:
@kwoodson I think this is a good suggestion.  I didn't enjoy writing that plugin ;)

I didn't even want to touch this portion of the code since this is a large patch set.  It's basically a drop-in replacement for what we were doing with facts + filter.

Yeah, let's do this in the next PR.